clients/admin-ui/src/features/common/custom-fields/CustomFieldsList.tsx

```diff
@@ -1,169 +1,41 @@
-      )!.allow_list_id,
-    }));
-
-    const unassigned =
-      customFieldDefinitions?.filter(
-        (item) =>
-          !withIds.some(
-            (existing) => existing.custom_field_definition_id === item.id
-          )
-      ) || [];
-
-    const withNoIds: CustomFieldWithIdExtended[] = [];
-    unassigned.forEach((item) => {
-      withNoIds.push({
-        allow_list_id: item.allow_list_id,
-        custom_field_definition_id: item.id as string,
-        resource_id: resourceId,
-        value: item.field_type === AllowedTypes.STRING ? "" : [],
-      });
-    });
-
-    return [...withIds, ...withNoIds];
-  }, [customFieldDefinitions, customFields, resourceId]);
-
-  const getCustomFieldDefinition = useCallback(
-    (item: CustomFieldWithIdExtended) =>
-      customFieldDefinitions!.find(
-        (cfd) => cfd.id === item.custom_field_definition_id
-      ),
-    [customFieldDefinitions]
-  );
-
-  const getListOptions = useCallback(
-    (allow_list_id: string) => {
-      const value = allAllowList?.find((item) => item.id === allow_list_id);
-      const list = value?.allowed_values!.map((item) => ({
-        value: item,
-        label: item,
-      }));
-      return list;
-    },
-    [allAllowList]
-  );
-
-  const handleSubmit = useCallback(
-    (values: FormValues) => {
-      try {
-        let hasError = false;
-        values.customFields.forEach(async (customField) => {
-          if (!customField.id && customField.value.length === 0) {
-            return;
-          }
-          const clone = { ...customField };
-          delete clone.allow_list_id;
-          const result = await upsertCustomField(clone);
-          if ("error" in result) {
-            hasError = true;
-          }
-        });
-        if (hasError) {
-          errorAlert(
-            `One or more custom field(s) has failed to save, please try again`
-          );
-        } else {
-          successAlert(
-            `Custom field(s) successfully saved and added to this ${resourceType} form`
-          );
-        }
-      } finally {
-        setSubmitting(false);
-      }
-    },
-    [errorAlert, resourceType, setSubmitting, successAlert, upsertCustomField]
-  );
-
-  useEffect(() => {
-    if ((isLoading && customFieldDefinitions) || customFields) {
-      // Update the initial parent customFields
-      (formValues as any).customFields = [...editCustomFieldList];
-      setIsLoading(false);
-    }
-  }, [
-    customFieldDefinitions,
-    customFields,
-    editCustomFieldList,
-    formValues,
+    definitionIdToCustomField,
+    idToAllowListWithOptions,
+    idToCustomFieldDefinition,
+    isEnabled,
     isLoading,
-  ]);
+    sortedCustomFieldDefinitionIds,
+  } = useCustomFields({
+    resourceFidesKey,
+    resourceType,
+  });
 
-  useEffect(() => {
-    if (isSubmitting && (formValues as any).customFields) {
-      handleSubmit(formValues as FormValues);
-    }
-  }, [formValues, handleSubmit, isSubmitting]);
```

Comment:
**Key change**
A major motivation for integrating custom fields onto the taxonomy hooks is to avoid this monkey patching of Formik's submission state. Formik has good handling of async (promise) submission, so moving the logic to a single chain of API calls removes the need to do complex state-timing debugging.